db/common_test.go

```diff
@@ -0,0 +1,170 @@
+		panic(err)
+	}
+}
+
+func (a *testArgs) setPsqlInfo() {
+	a.psqlInfo = fmt.Sprintf("host=%s port=%d user=%s "+
+		"password=%s dbname=%s sslmode=disable",
+		a.host, a.port, a.user, a.password, a.dbname)
+	a.logger.Info(a.psqlInfo)
+}
+
+func makeUUID() (r string) {
+	b := make([]byte, 16)
+	if _, err := rand.Read(b); err != nil {
+		panic(err)
+	}
+	uuid := fmt.Sprintf("%x-%x-%x-%x-%x",
+		b[0:4], b[4:6], b[6:8], b[8:10], b[10:])
+	return uuid
+}
+
+func newTestParams() *testArgs {
+	var err error
+	a := &testArgs{
+		host:     "localhost",
+		port:     5432,
+		tinkDb:   nil,
+		user:     "tinkerbell",
+		password: "tinkerbell",
+		dbname:   "tinkerbell",
+		workflowData: Workflow{
+			State: 0,
+			ID:    `b356eede-22c1-11eb-8c17-0242ac120004`,
+			Hardware: `{
+				"id": "ce2e62ed-826f-4485-a39f-a82bb74338e2",
+				"metadata": {
+					"facility": {
+						"facility_code": "onprem"
+					},
+					"instance": {
+						"ip_addresses": [{
+							"address": ""
+						}]
+					},
+					"state": ""
+				},
+				"network": {
+					"interfaces": [{
+						"dhcp": {
+							"arch": "x86_64",
+							"ip": {
+								"address": "192.168.1.5",
+								"gateway": "192.168.1.1",
+								"netmask": "255.255.255.248"
+							},
+							"mac": "08:00:27:00:00:01",
+							"uefi": false
+						},
+						"netboot": {
+							"allow_pxe": true,
+							"allow_workflow": true
+						}
+					}]
+				}
+			}`,
+			Template: `da013837-a30c-237a-adfc-2d9ad69d5011`,
+		},
+		hardwareData: `{
+			"id": "ce2e62ed-826f-4485-a39f-a82bb74338e2",
+			"metadata": {
+				"facility": {
+					"facility_code": "onprem"
+				},
+				"state": ""
+			},
+			"network": {
+				"interfaces": [{
+					"dhcp": {
+						"arch": "x86_64",
+						"ip": {
+							"address": "192.168.1.5",
+							"gateway": "192.168.1.1",
+							"netmask": "255.255.255.248"
+						},
+						"mac": "08:00:27:00:00:01",
+						"uefi": false
+					},
+					"netboot": {
+						"allow_pxe": true,
+						"allow_workflow": true
+					}
+				}]
+			}
+		}`,
+		instanceData: ` {
+		"id": "da013837-a30c-237a-adfc-2d9ad69d5000",
+		"instance": {
+		  "ip_addresses": [
+			{
+			  "address": "192.168.0.1"
+			}
+		  ]
+		}
+	  }`,
+		templateData: `
+version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+- name: "hello world"
+  worker: "192.168.0.1"
+  actions:
+  - name: "hello_world"
+    image: hello-world
+    timeout: 60`,
+	}
+	////////////////////////////////////////////////////////////////////////////
+	a.setLogger()
+	a.setPsqlInfo()
+	a.setCTX()
+	a.tinkDb = Connect(a.logger)
+	if a.tinkDb.instance, err = sql.Open("postgres", a.psqlInfo); err != nil {
+		panic(err)
+	}
+	return a
+}
+
+// TestConnect : validates database connection.
+func TestConnect(t *testing.T) {
+	var db *sql.DB
+	var err error
+	a := newTestParams()
+	a.setLogger()
+	a.setPsqlInfo()
+	if db, err = sql.Open("postgres", a.psqlInfo); err != nil {
+		t.Fail()
+	}
```

Comment:
This is already being done in newTestParams, do we still need it here?

Reply:
Removed.